DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionSwizzler.swift

```diff
@@ -6,74 +6,102 @@
 
 import Foundation
 
-/// Swizzles `URLSession` methods.
-internal class URLSessionSwizzler {
-    private static var _dataTaskWithURLRequestAndCompletion: DataTaskWithURLRequestAndCompletion?
-    static var dataTaskWithURLRequestAndCompletion: DataTaskWithURLRequestAndCompletion? {
-        get {
-            lock.lock()
-            defer { lock.unlock() }
-            return _dataTaskWithURLRequestAndCompletion
-        }
-        set {
-            lock.lock()
-            defer { lock.unlock() }
-            _dataTaskWithURLRequestAndCompletion = newValue
-        }
-    }
+/// Swizzles `URLSession*` methods.
+internal final class URLSessionSwizzler {
```

Comment:
/Blocking

Combining all the swizzling code inside one type was never the intent but I understand the reasoning behind. However, this can be done with the existing approach by making the each swizzler type as intance type.

This makes the testing each swizzling class easy, better code navigation. You can compose those types in a `NetworkInstrumentationSwizzler` type. 

something like
```swift
class NetworkInstrumentationSwizzler {
    let urlSessionSwizzler: URLSessionSwizzler
    let urlSessionTaskDelegateSwizzler: URLSessionTaskDelegateSwizzler
    ....
}
```

Reply:
As suggested, I have split the swizzler class, see the last commit. But TBH, I don't see the benefit 🤔 It just adds another layer and doesn't help much with code navigation.